### PR TITLE
Update examples.jl

### PR DESCRIPTION
**Pull request by yakir12**
Added an example for a `heatmap` with a `DateTime` axis.

### CODE REVIEW

**Comment by mkborregaard** [comment]
Nice - this will depend on resolving #1283 , right?

The way Plots testing works is that it runs the examples and compares it to a repo of sample figures (clonable as a package called PlotReferenceImages.jl).

@yakir12 when the issue's been resolved and this won't break tests, could I ask you to checkout PlotReferenceImages.jl, and run the test in runtests.jl manually with this PR - that should put the correct image in the PlotReferenceImages folder, which can then be PR'ed on that repo. Thanks!

**Comment by yakir12** [comment]
Sure. I'll wait for #1283

**Comment by daschw** [comment]
#1283 should now be fixed.

**Comment by mkborregaard** [comment]
bump @yakir12 ?

**Comment by yakir12** [comment]
Hi! Sorry for the loooooooong delay. I finally got to it, but it seems that #1283 didn't fix it:
```julia
Error showing value of type Plots.Plot{Plots.GRBackend}:
ERROR: MethodError: no method matching convert_sci_unicode(::Date)
Closest candidates are:
  convert_sci_unicode(::AbstractString) at /home/yakir/.julia/dev/Plots/src/utils.jl:1112
Stacktrace:
 [1] gr_display(::Plots.Subplot{Plots.GRBackend}, ::Measures.Length{:mm,Float64}, ::Measures.Length{:mm,Float64}, ::Array{Float64,1}) at /home/yakir/.julia/dev/Plots/src/backends/gr.jl:880
 [2] gr_display(::Plots.Plot{Plots.GRBackend}, ::String) at /home/yakir/.julia/dev/Plots/src/backends/gr.jl:585
 [3] gr_display at /home/yakir/.julia/dev/Plots/src/backends/gr.jl:543 [inlined]
 [4] _display(::Plots.Plot{Plots.GRBackend}) at /home/yakir/.julia/dev/Plots/src/backends/gr.jl:1437
 [5] display(::Plots.PlotsDisplay, ::Plots.Plot{Plots.GRBackend}) at /home/yakir/.julia/dev/Plots/src/output.jl:145
 [6] display(::Any) at ./multimedia.jl:287
 [7] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [8] invokelatest at ./essentials.jl:696 [inlined]
 [9] print_response(::IO, ::Any, ::Any, ::Bool, ::Bool, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:154
 [10] print_response(::REPL.AbstractREPL, ::Any, ::Any, ::Bool, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:139
 [11] (::getfield(REPL, Symbol("#do_respond#40")){Bool,getfield(REPL, Symbol("##50#59")){REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:708
 [12] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [13] invokelatest at ./essentials.jl:696 [inlined]
 [14] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/LineEdit.jl:2261
 [15] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:1029
 [16] run_repl(::REPL.AbstractREPL, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:191
 [17] (::getfield(Base, Symbol("##719#721")){Bool,Bool,Bool,Bool})(::Module) at ./logging.jl:311
 [18] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [19] invokelatest at ./essentials.jl:696 [inlined]
 [20] macro expansion at ./logging.jl:308 [inlined]
 [21] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:330
 [22] exec_options(::Base.JLOptions) at ./client.jl:242
 [23] _start() at ./client.jl:421
```

**Comment by ssfrr** [comment]
I just ran into the same error trying to plot a heatmap with a DateTime axis. Any workarounds?

**Comment by ssfrr** [comment]
Adding the argument `xformatter=Plots.dateformatter` to the plot call fixes the issue.

File: src/examples.jl [hunk 1]
```diff
@@ -415,6 +415,17 @@ attribute. The default framestyle is `:axes`.
...
 ),
+PlotExample("Heatmap with DateTime",
+"""
+Parse `DateTime` correctly when used in the axes of a `heatmap` plot.
+""",
+    [:(begin
+    x = DateTime(2001):Dates.Year(1):DateTime(2005)
+    y = 1:5
+    z = rand(5,5)
+    heatmap(x, y, z, title = "Heatmap with DateTime")
```
Comment (asinghvi17) [inline src/examples.jl]:
```suggestion
    heatmap(x, y, z, title = "Heatmap with DateTime", xformatter = Plots.dateformatter)

**Comment by asinghvi17** [comment]
Thanks!  @daschw is this OK to merge?

**Comment by asinghvi17** [comment]
It's worth noting that this fails on `plotly`:
```julia
heatmap(x, y, z, title = "Heatmap with DateTime", xformatter = Plots.dateformatter, xrotation = 90)
Error showing value of type Plots.Plot{Plots.PlotlyBackend}:
ERROR: MethodError: no method matching length(::DateTime)
Closest candidates are:
  length(::Core.SimpleVector) at essentials.jl:597
  length(::Base.MethodList) at reflection.jl:819
  length(::Core.MethodTable) at reflection.jl:893
```

**Comment by BeastyBlacksmith** [comment]
This seems related to https://github.com/JuliaPlots/Plots.jl/pull/2351

**Pull request closed without merging**